### ISSUE TITLE
Ray Renderer?

### ISSUE THREAD

**Issue by bolopenguin** [nerfstudio-project/gsplat#437]
Hello!

I was wondering it could be possible to have a ray renderer, or a workaround.
For example, given the ray origins and directions get the color seen from that ray.

Thanks you very much!

**Comment by maturk** [comment]
Yes it should be possible, there are analytical formulas for ray-Gaussian intersections, that could be implemented. I imagine this could be quite slow, if naively done in pytorch. 

Gojic's team have a nice paper https://gaussiantracer.github.io speeding it up with a BVH and other tricks. Are you interested in working on the problem? There is an earlier project from [Leonid Keselman](https://leonidk.com/fmb-plus/) that does it as well, and the python code is available (also in nerfstudio [here](https://github.com/leonidk/ns-fmb)). The main thing is just implementing this equation:
<img width="773" alt="image" src="https://github.com/user-attachments/assets/300d12be-201b-47f1-a0ec-2417d1651aa5">
Would be interesting to see if some ideas from the Gojic paper could be incorporated to gsplat.
